LAGJH-1047 CRUD for Headings.

little9 opened the pull request:
This adds CRUD for Headings.

This PR has additional database migrations. After
checking out this branch you'll need to run
docker-compose run --rm hummingbird bundle exec rake db:migrate
to update the database.

You can test the functionality by logging in and visiting
/headings/, /subheadigns and /database_headings.

You'll need to create a heading and a subheading. After that
you'll be able to create a database_heading, which associates
a heading with a database.

These are the most basic forms for this functionality. We'll
need to combine some of the forms so that you don't need to
visit the /database_headings forms to add a heading to a database.

little9 commented:
@jatjhu @jhu-alistair I added some spacing to the forms (like Stephen suggested on the last PR) and buttons for the different content types on the home page. Should make testing a little easier!

jatjhu reviewed (approved):
I pulled down you changes and tested them, it looks good to me. The code looks good and the new forms look great!

jhu-alistair commented:
- [ ] Add DB
- [ ] Add multiple Headings
- [ ] Add multiple Subheadings
- [ ] Change Subheading label
- [ ] Change Heading label

jhu-alistair commented:
Enhancement:  enforce unique combination of JHUID + Heading ID + Subheading ID so that we never get duplicate entries.

little9 commented:
@jhu-alistair I added the multicolumn index for database_headings and the unique label index for headings/subheadings. 

In my testing, if I deleted a database with an associated database_heading the database_headings was also deleted. 

You can pull in these db changes by pulling in the changes on this branch and running `docker-compose run --rm hummingbird bundle exec rails db:migrate`.

jhu-alistair commented:
I managed to add a heading with an empty label. See my new comments on https://jhulibraries.atlassian.net/browse/LAGJH-1054
